enterprise/models.py

```diff
@@ -4259,11 +4260,34 @@ class Meta:
         unique_together = (("name", "enterprise_customer"),)
         ordering = ['-modified']
 
-    def get_all_learners(self):
+    def get_all_learners(self, q=None, sort_by=None):
         """
         Returns all users associated with the group, whether the group specifies the entire org else all associated
         membership records.
-        """
+
+        Params:
+            q (optional): filter the returned members list by user email and name with a provided sub-string
+            sort_by (optional): specify how the list of returned members should be ordered. Supported sorting values
+            are `memberDetails`, `memberStatus`, and `recentAction`. Ordering can be reversed by supplying a `-` at the
+            beginning of the sorting value ie `-memberStatus`.
+        """
+        if q:
+            # Unfortunately, user (and by extension email) is a property not a field on the ecu model. In order to
+            # filter ecu's by email,  we have to select from the Users table. To make this efficient, we join the user
+            # table on ecu table where the ecu is associated with the enterprise customer on the group. This limits how
+            # many User records we're calling `where email like %` on.
+            # Parameterizing the query makes this raw sql safe from sql injects re the Django docs-
+            # https://docs.djangoproject.com/en/5.0/topics/security/
+            var_q = f"%{q}%"
+            sql_string = """
+                select entcu.id from enterprise_enterprisecustomeruser entcu
+                join auth_user au on entcu.user_id = au.id
+                where entcu.enterprise_customer_id = %s and au.email like %s;
+            """
+            # Raw sql is picky about uuid format
+            customer_id = str(self.enterprise_customer.pk).replace("-", "")
+            ecus = EnterpriseCustomerUser.objects.raw(sql_string, (customer_id, var_q))
+            ecus_ids_list = [ecu.id for ecu in ecus]
         if self.applies_to_all_contexts:
             members = []
```

Comment:
Also, this whole chunk could be something like `def _get_implicit_group_members(self, user_query=None, ecu_ids=None)`.